ui/raidboss/popup-text.ts

```diff
@@ -380,17 +380,55 @@ class TriggerOutputProxy {
     }
 
     return value.replace(/\${\s*([^}\s]+)\s*}/g, (_fullMatch: string, key: string) => {
-      if (params !== undefined && key in params) {
-        const str = params[key];
-        switch (typeof str) {
-          case 'string':
-            return str;
-          case 'number':
-            return str.toString();
+      if (params !== undefined) {
+        let prop: string | undefined = undefined;
+
+        if (!(key in params) && key.includes('.')) {
+          const parts = key.split('.');
+          key = parts[0] ?? '';
+          prop = parts[1];
+        }
+
+        if (key in params) {
+          const val = params[key];
+          switch (typeof val) {
+            case 'string':
+              return val;
+            case 'number':
+              return val.toString();
+            case 'object':
+              if (val !== null) {
+                const looseVal = val as { [outputName: string]: unknown };
```

Comment:
I think this `as` is only needed because you're using `object` as a type (which we probably should disallow in eslint).  I think if it were some sort of `{ [key: string]: string }` or even `{ [key: string]: unknown }` this code would be more straightforward.